peura/src/Peura/GHC.hs

```diff
@@ -27,9 +29,11 @@ data GhcInfo = GhcInfo
 
 getGhcInfo :: FilePath -> Peu r GhcInfo
 getGhcInfo ghc = do
-    ghcDir   <- getAppUserDataDirectory "ghc"
+    ghcDir  <- getAppUserDataDirectory "ghc"
+    mbGhcExe  <- liftIO $ D.findExecutable (addExeExtension ghc)
```

Comment:
AFAICS from `directory` documentation:

> Given a name or path, `findExecutable` appends the `exeExtension` to the query and searches for executable files

Though it's under `findExecutablesInDirectories`, by looking at the code, `findExecutable` uses `findExecutablesLazyInternal` which on Windows calls to [`Win32.searchPath`](https://hackage.haskell.org/package/Win32-2.8.5.0/docs/System-Win32-Info.html#v:searchPath) which is a wrapper over https://docs.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-searchpathw which says for extension argument

> `lpExtension`
>
> The extension to be added to the file name when searching for the file. The first character of the file name extension must be a period (.). The extension is added only if the specified file name does not end with an extension.
>
> If a file name extension is not required or if the file name contains an extension, this parameter can be NULL.

Without the windows specific `exe` extension hacks, this patch would be an improvement.

---

If this works as I think it should, i.e. without manual extension appending could you also make a PR to improve `directory` documentation

Reply:
Mmm i tested the behaviour of `System.Directory.findExecutable` and is quiet weird:
```
*CabalStoreCheck.Main> :m + System.Directory
*CabalStoreCheck.Main System.Directory> findExecutable "ghc"
Just "D:\bin\\Programs\\stack\\x86_64-windows\\ghc-8.6.5\\bin\\ghc.exe"
*CabalStoreCheck.Main System.Directory> findExecutable "ghc-8.6.5"
Nothing
*CabalStoreCheck.Main System.Directory> findExecutable "D:\\bin\\Programs\\stack\\x86_64-windows\\ghc-8.6.5\\bin\\ghc-8.6.5"
Just "D:\\bin\\Programs\\stack\\x86_64-windows\\ghc-8.6.5\\bin\\ghc-8.6.5.exe"
```
So it seems that if the executable string is not a path *and* it has an extension (like `.6.5` but whatever) it is not able to find it:
```
*CabalStoreCheck.Main System.Directory> findExecutable "D:\\Trabajo\\NO-SALVAR\\jneira\\bin\\Programs\\stack\\x86_64-windows\\ghc-8.6.5\\bin\\ghc.ext"
Just "D:\\Trabajo\\NO-SALVAR\\jneira\\bin\\Programs\\stack\\x86_64-windows\\ghc-8.6.5\\bin\\ghc.ext.exe"
*CabalStoreCheck.Main System.Directory> findExecutable "ghc.ext"
Nothing
*CabalStoreCheck.Main System.Directory> findExecutable "ghc.ext.exe"
Just "D:\\Trabajo\\NO-SALVAR\\jneira\\bin\\Programs\\stack\\x86_64-windows\\ghc-8.6.5\\bin\\ghc.ext.exe"
```

After checking the `findExecutable` behaviour i could not do anything other than add the windows hack :disappointed: